syntax = "proto3";

package image;

service Images {
    rpc buildImage(Image) returns (stream ImageBuildResponse) {}
    rpc pushImage(ImageRegistryDetails) returns (GenericResponse) {}
}

message Image {
    string name = 1;
    bytes dockerfile = 2;
    string tag = 3;
}

message Error {
    string type = 1;
    string error = 2;
}

message Success {
    string type = 1;
    string msg = 2;
}

message ImageBuildSuccess {
    string name = 1;
    string logs = 2;
}

message ImageBuildResponse {
    oneof responses {
        ImageBuildSuccess success = 1;
        Error error = 2;
    }
}

message ImageRegistryDetails {
    string name = 1;
    string tag = 2;
    string hub = 3;
    string credential = 4;
}

message GenericResponse {
    oneof responses {
        Success success = 1;
        Error error = 2;
    }
}
